### PR TITLE
fix sonarlint issues and reduce false positives

### PR DESCRIPTION
**Pull request by Bertk**
- reduce SonarQube issues (potential bugs)
- mark some issues as false positives in code

---
This change is [Reviewable](https://reviewable.io/reviews/sonaropencommunity/sonar-cxx/1114)

### CODE REVIEW

File: cxx-sensors/src/main/java/org/sonar/cxx/sensors/drmemory/CxxDrMemorySensor.java [hunk 1]
```diff
@@ -85,15 +91,6 @@ protected void processReport(final SensorContext context, File report) {
           }
         }
       }
-    } catch (IOException e) {
```
Comment (guwirth) [inline cxx-sensors/src/main/java/org/sonar/cxx/sensors/drmemory/CxxDrMemorySensor.java]: @Bertk why is this removed? Should break or continue (handling inside validateRecovery).

**Comment by guwirth** [comment]
@Bertk can you have a look to the comment above please.

File: cxx-sensors/src/main/java/org/sonar/cxx/sensors/drmemory/CxxDrMemorySensor.java (hunk 1 above)
Comment (Bertk) [inline cxx-sensors/src/main/java/org/sonar/cxx/sensors/drmemory/CxxDrMemorySensor.java]: The only IO operations in this scope are handled with a try..catch block in ```org.sonar.cxx.sensors.utils.normalizePath(String filename)``` and ```getElements(file, charset)```.

I have also some doubts with the validateRecovery in this context. Whenever a report cannot be parsed (IOException) the analysis should not be stopped. This is the current behavior for other tools with reliability issues like Vera++, PC-lint and RATS.

**Comment by guwirth** [comment]
@Bertk yes you are right.
